language/move-lang/src/lib.rs

```diff
@@ -180,6 +180,41 @@ pub fn move_continue_up_to(pass: PassResult, until: Pass) -> Result<PassResult,
     run(pass, until)
 }
 
+/// Similar to `move_compile`, but with an additional `allow_shadow` flag to indicate that modules
+/// defined in targets can shadow (i.e., override) the modules in dependencies. In other words,
+/// if the same module id (<address::name>) is found in both targets and dependencies, the module
+/// definition in targets takes priority.
+///
+/// This function also allows to specify a stopping point of the compilation process. Similar to
+/// `move_continue_up_to`, the stopping point is inclusive, meaning that the pass specified by
+/// `until: Pass` will be run
+pub fn move_compile_up_to(
+    targets: &[String],
+    deps: &[String],
+    sender_opt: Option<Address>,
+    interface_files_dir_opt: Option<String>,
+    allow_shadow: bool,
+    until: Pass,
+) -> anyhow::Result<(FilesSourceText, Result<PassResult, Errors>)> {
+    let (files, pprog_and_comments_res) =
+        move_parse(targets, deps, sender_opt, interface_files_dir_opt)?;
+    let (_comments, sender_opt, pprog) = match pprog_and_comments_res {
+        Err(errors) => return Ok((files, Err(errors))),
+        Ok(res) => res,
+    };
+
+    let pprog = if allow_shadow {
+        shadow_lib_module_definitions(pprog, sender_opt)
+    } else {
+        pprog
+    };
+
+    Ok((
+        files,
+        move_continue_up_to(PassResult::Parser(sender_opt, pprog), until),
+    ))
+}
+
```

Comment:
I just went through a big refactor trying to slim this down, so I'm not really keen on adding another entry point. 

I think `sources_shadow_deps` should just be a flag on all compiler entry points

Reply:
Just a heads up, this will lead to many changes across the `language/` dir if every entry points of the compiler need to be updated...